src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/AttributeRouteModel.cs

```diff
@@ -99,6 +99,33 @@ public bool IsAbsoluteTemplate
             };
         }
 
+        /// <summary>
+        /// Combines the prefix and route template for an attribute route.
+        /// </summary>
+        /// <param name="prefix">The prefix.</param>
+        /// <param name="template">The route template.</param>
+        /// <returns>The combined pattern.</returns>
+        public static string CombineTemplates(string prefix, string template)
```

Comment:
See comment below, we shouldn't reuse this for razor pages. No need to add more public API surface for this